### PR TITLE
Noskip testing. fixes #12

### LINKED ISSUE #12: Test skips makes it possible to accidentally merge something that doesn't run the tests
I am wondering: Does having this check in place make it possible to accidentally merge something that doesn't run the tests? Imagine the following:

* I update the container, and accidentally remove the model path
* Tests are run, but since ROOTPATH = None, it will simply skip the tests
* GitHub Actions says that all is ✔️ , without mentioning the skippet tests unless I check the logs (unlikely!)

I get that you want to make it easy during development, but I wonder if it is better to be honest to the local developer and failing the tests because the testdata has not been configured.

_Originally posted by @tankefugl in https://github.com/SINTEF/PySilCam/pull/1_

Additional information:
https://pypi.org/project/pytest-error-for-skips/ may be what we are looking at, forcing all skips to be error on the CI, while allowing skips on other systems.

### PR DESCRIPTION
**Pull request by emlynjdavies**
Running python setup.py test_noskip will now cause an error if a test is skipped. This is implemented in the Dockerfile so that guthub actions require all tests to pass.

python setup.py test is unaffected.

### CODE REVIEW

**Comment by emlynjdavies** [comment]
I don't understand why none of the Actions are working (also on other branches).

**Comment by emlynjdavies** [comment]
> @tankefugl I don't understand why none of the Actions are working (also on other branches).

Seems they are working again now. maybe something was wrong on the github end?

File: setup.py [hunk 1]
```diff
@@ -5,6 +5,22 @@
...
+class PyTestNoSkip(TestCommand):
+    def finalize_options(self):
+        TestCommand.finalize_options(self)
+        # use pytest plugin to force error if a test is skipped
+        self.test_args = ['--error-for-skips']
+        self.test_suite = True
+
+    def run_tests(self):
+        import pytest
+        params = {"args":self.test_args}
```
Comment (tankefugl) [inline setup.py:18]: E231: missing whitespace after ':'

File: setup.py [hunk 2]
```diff
@@ -5,6 +5,22 @@
...
+        params["args"] +=  ["--junitxml", "test-report/output.xml"]
```
Comment (tankefugl) [inline setup.py:19]: E222: multiple spaces after operator

File: environment.yml [hunk 3]
```diff
@@ -32,3 +32,4 @@ dependencies:    # These are the conda packages
         - tflearn==0.3.2
         - PyQt5==5.10
         - cmocean==1.2
+        - pytest-error-for-skips
```
Comment (tankefugl) [inline environment.yml:35]: Consider adding an explicit version for pytest-error-for-skips

Comment (tankefugl) [inline environment.yml:35]:
```suggestion
        - pytest-error-for-skips=2.0.2
```

**Pull request merged**